docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,263 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream methods for downloading files, it is time to introduce you to additional *upstream* methods. Upstream methods refers to functions that publish files to the Hub from your library. This guide will show you how to:
+
+* Use the `HfApi` class to manage a repository.
+* Use the `Repository` class to manage a repository with Git-like commands.
```

Comment:
These two bullet points are clashing a bit, it looks like `Repository` is a different approach to what `HfApi` is already doing.
How about the following (feel free to adapt the wording):
```suggestion
* Use the `HfApi` class to manage a repository.
* Use the `Repository` class to handle files and version control of your repositories using git-like commands.
```

Reply:
I agree, this is a more precise description of `Repository` :)